test/Autofac.Specification.Test/Registration/RegistrationOnlyIfTests.cs

```diff
@@ -62,6 +83,123 @@ public void IfNotRegistered_CanBeDecorated()
             Assert.IsType<Decorator>(result);
         }
 
+        [Fact]
+        public void IfNotRegistered_CanConditionGenericService()
+        {
+            var builder = new ContainerBuilder();
+            builder.RegisterGeneric(typeof(SimpleGeneric<>)).IfNotRegistered(typeof(SimpleGeneric<int>));
+            var container = builder.Build();
+
+            var result = container.Resolve<SimpleGeneric<int>>();
+            Assert.IsType<SimpleGeneric<int>>(result);
+        }
+
+        [Fact]
+        public void IfNotRegistered_CanConditionGenericServiceWithMultipleInterfaces()
+        {
+            var builder = new ContainerBuilder();
+            builder.RegisterGeneric(typeof(MultiServiceGeneric<>)).As(typeof(IService<>)).As(typeof(IService2<>));
+            builder.RegisterGeneric(typeof(MultiServiceGeneric<>)).As(typeof(IService<>)).IfNotRegistered(typeof(IService2<int>));
```

Comment:
I think this would be more clear if two different open generics were being registered. Regardless of whether `IfNotRegistered` works, `IService<>` will be a `MultiServiceGeneric<>`.